Marlin/src/HAL/STM32F1/tft/tft_fsmc.h

```diff
@@ -30,6 +30,11 @@
 
 #include <libmaple/dma.h>
 
+#undef FSMC_DMA_DEV
+#undef FSMC_DMA_CHANNEL
+#define FSMC_DMA_DEV                      DMA1
+#define FSMC_DMA_CHANNEL               DMA_CH1
```

Comment:
i'm not sure its a good idea to hardcode that without the ability to change it...

Reply:
These parameters were for STM32F1 HAL only and they were the same on all boards, so it does not look like they are needed as configurable parameters.
Copy-paste method even brought them to `stm32f4/pins_TRONXY_V10.h` where they could not possibly be used.
STM32 HAL do not use them at all, so I feel like it is time to get rid of them and move to unified configuration.

Reply:
but maple is not F4 related, there was attempt to support first F4 chips but in marlin its for the "stable" F1 chips

Reply:
this include is maple related... no reason to break it... this HAL STM32F1 is maple...

Reply:
maple should be faded out to improve the maintainability of this voluntary project. Additional code leads to burden. We are looking at ways to create an even better firmware and maple is in the way.

Reply:
Maple may be deprecated, but support has not removed yet (and likely won't be for a while now that GD32 support will likely be merged soon).

We'll still have to keep it working for now.

Reply:
Please note that @thisiskeithb probably means using maple as a backup for legal reasons. I have already heard that argument from @ellensp . There should be no direct correlation between maple and GD32. If that company did their job properly then GD32 would have it's own HAL.

Reply:
@tpruvot, I think we have some misunderstanding here, let me try to elaborate my point.

Prior to this pull request:
STM32F1 maple HAL have an option to configure DMA controller and DMA channel for FSMC data transfer.
All boards have the very same DMA2_CH5 configuration in their pins.h files. The customization functionality is not used.

STM32 HAL have FSMC DMA controller and channel hardcoded and defined by MCU family - DMA2_CH1 for F1.

Someone copied these maple-specific settings to `stm32f4/pins_TRONXY_V10.h` file not knowing that these settings have no effect on STM32 HAL.

Here is how I see it:
 - we have configurable parameters that are not board-specific, but rather MCU specific
 - I can't think of a reasonable scenario when these DMA parameters need to be changed for a specific board
 - it is possible to ruin your build if you set DMA channel to the one used by SDIO (DMA2_CH4)

From where I stand there is no real value of having these as a configurable parameters, but they do cause confusion (`pins_TRONXY_V10.h`) and have to be defined in each (maple) FSMC-capable board's pins.h file. 
New settings were tested on both F103VE and F103ZE MCUs and they have certain advantage over existing settings.
I would rather use them globally and for both HALs than make them configurable.
Do you know of a use case where these needs to be configured differently?

Reply:
@jmz52 sounds like he thinks people desperate need that configuration, even not giving a solid example... I am interested aswell, jmz52. From my POV your changes do not break anything but imagination of people involved.